analyses/mutational-signatures/mutational_signatures.Rmd

```diff
@@ -368,6 +369,247 @@ lapply(unique(nature_sigs_primary$short_histology),
 )
 ```
 
+## Determine Signatures de novo  
+This section is included, because the previous strategies imply that there are mutagenic mechanisms in operation that are unlikely to be a major factor in this dataset, such as smoking and UV light.
+This is happening due to overfitting problems when fitting existing signatures.  De novo signature calling is used to provide an alternative for discussion. It is determined, which signatures can be extracted from the data themselves. These signatures are then fitted back to the known signatures that are used for the fitting approach above. 
+
+### De novo calling
+
+To prevent the loading of big data files into memory and intense processing, I have muted this part of the script and it loads the preprocessed parameters. 
+```{r}
+# determine the ideal number of signatures. 
+# the choice of 10 signatures is now hard coded, because otherwise it has to be rerun.
+# this step requires extensive ressources, the signature file is > 8 GB in size. Therefore a loop is included to load a preprocessed file, if available. 
+
+if(!file.exists("results/sigfit_signatures_10.rdat")){
+  
+  samples_extr <- extract_signatures(counts = sigs_input,
+                                        nsignatures = 5:15,
+                                        iter = 1000, 
+                                        seed = 42)
+  dev.off()
+  
+  samples_extr10 <- extract_signatures(counts = sigs_input,
+                                        nsignatures = 10,
+                                        iter = 10000, 
+                                        seed = 42)
+    
+  save(samples_extr10, file="results/sigfit_signatures_10.rdat")
+}
+```
+
+```{r}
+if(!file.exists("results/sigfit_signatures_10_signatures.rdat")){
```

Comment:
I am unfamiliar with the `.rdat` file type - do you know how it compares to `.RDS` and `.RData`?

Reply:
I have learned to use .rdat for saving R objects. I guess it is the same as .RData.